components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -512,22 +520,92 @@ export default class PluginManagement extends AdminSettings {
         Utils.clearFileInput(element);
     }
 
-    handleOverwritePluginCancel = () => {
+    handleOverwriteUploadPluginCancel = () => {
         this.setState({
             file: null,
             fileSelected: false,
             serverError: null,
-            confirmModal: false,
+            confirmOverwriteUploadModal: false,
             lastMessage: null,
             uploading: false,
         });
     }
 
-    handleOverwritePlugin = () => {
-        this.setState({confirmModal: false});
+    handleOverwriteUploadPlugin = () => {
+        this.setState({confirmOverwriteUploadModal: false});
         this.helpSubmitUpload(this.state.file, true);
     }
 
+    onPluginDownloadUrlChange = (e) => {
+        this.setState({
+            pluginDownloadUrl: e.target.value,
+        });
+    }
+
+    helpSubmitInstall = async (force) => {
```

Comment:
maybe just `installFromURL`? Not a fan of having `help` in helper function names.